hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/metrics/FlinkCompactionMetrics.java

```diff
@@ -103,4 +110,16 @@ public void endCompaction() {
     this.compactionCost = stopTimer(COMPACTION_KEY);
   }
 
+  public void setCompactionFailedState(CompactionState compactionState){
+    this.compactionFailedState = compactionState.state;
```

Comment:
Mayne we have 2 methods here:

`markCompactionCompleted` and `markCompactionRolledBack`.